robot/packages/cloud_package_manager.go

```diff
@@ -507,21 +512,17 @@ func (m *cloudManager) unpackFile(ctx context.Context, fromFile, toDir string) e
 
 	// Now we make another pass creating the links
 	for i := range links {
-		select {
-		case <-ctx.Done():
-			return errors.New("interrupted")
-		default:
+		if err := ctx.Err(); err != nil {
+			return err
 		}
 		if err := linkFile(links[i].Name, links[i].Path); err != nil {
 			return errors.Wrapf(err, "failed to create link %s", links[i].Path)
 		}
 	}
 
 	for i := range symlinks {
```

Comment:
[q] - Do you know why we do links and symlinks separately when they basically do the same thing? Do we need to create these symlinks for existing files before we do symlinks? 

(this is for my understanding)

Reply:
Nope -- probably some ordering issue like a symlink can link to a link or something